core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -83,6 +89,11 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
   def main(args: Array[String]) {
 
     info("Starting mirror maker")
+    println("WARNING: The default partition assignment strategy of mirror maker will change from 'range' to 'roundrobin' " +
```

Comment:
@hachikuji I've been looking more closely into MM today to better understand your comments. I have to admit MM is not one of the best and most up-to-date documented features of Kafka (for example, I could not find anything concrete about how to make MM produce an identical cluster with same partitions and messages).

But from what I understand so far,

1. Ideally, MM should run continuously. If it stops and resumes there is a chance for duplicate / lost messages.
2. I don't see a way for consumers in MM to follow different config; e.g. use different assignment strategy. All consumers follow the same strategy (the default or what's given in consumer config). In other words, each MM instance runs with a single partition assignment strategy.
3. In a running MM consumers cannot come and go, and so MM uses a fixed config throughout execution.

With my understanding above I'm having a hard time realizing the compatibility issue, and why switching from one default strategy to another could be problematic compatibility-wise. That's why I'm guessing I've got something wrong above (for example, my second bullet above seems to be in conflict with your statement about enabling both strategies). In either case I'd appreciate your clarification.